src/hacl/hkdf.rs

```diff
@@ -0,0 +1,150 @@
+/// HKDF Errors.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Error {
+    /// The requested output key material in expand was too large for the used
+    /// hash function.
+    OkmTooLarge,
+    /// At least one function argument has been too large to process.
+    ArgumentsTooLarge,
+}
+
+macro_rules! impl_hkdf {
+    ($name:ident,$extract:ident,$expand:ident,$tag_len:literal) => {
+        pub mod $name {
+            use super::Error;
+
+            /// HKDF extract using the `salt`, and the input key material `ikm`.
+            /// Returns the pre-key material in an array of tag length.
+            ///
+            /// Note that this function panics if `salt` or `ikm` is larger than 2**32 bytes.
+            pub fn extract(salt: &[u8], ikm: &[u8]) -> [u8; $tag_len] {
+                let mut prk = [0u8; $tag_len];
+                unsafe {
+                    libcrux_hacl::$extract(
+                        prk.as_mut_ptr(),
+                        salt.as_ptr() as _,
+                        salt.len().try_into().unwrap(),
+                        ikm.as_ptr() as _,
+                        ikm.len().try_into().unwrap(),
+                    );
+                }
+                prk
+            }
+
+            /// HKDF expand using the pre-key material `prk` and `info`. The output length
+            /// is defined through the result type.
+            /// Returns the key material in an array of length `okm_len` or
+            /// [`Error::OkmTooLarge`] if the requested `okm_len` is too large.
+            ///
+            /// Note that this function returns an [`Error::ArgumentsTooLarge`]
+            /// if `salt`, `ikm`, or `OKM_LEN` is larger than 2**32 bytes.
+            pub fn expand<const OKM_LEN: usize>(
```

Comment:
The top level should probably expose something like this.
With all the warnings in here. We should make sure the top level APIs are done well and offer these different options, and then drop the functions here that are not necessary.
You can do that here, or file a follow-up.

Reply:
Yeah. Let's keep this PR for hack package utilization and file a follow-up for other changes.